mantidproject/mantid
# Plot spherical integrated peaks using ellipsoid representation

**Description of work.**

Removed the code specific to plotting integrated spheres (i.e. class `SphericalIntergratedPeakRepresentation`) and the associated tests. As spheres are a specific instance of an ellipsoids, the plotting of spheres can be handled by the same code as for ellipsoids. To do this I have written a fucntion ro convert the JSON dict of the sphere shape to a compatible ellipsoid. The advantage of doing this (other than removing similar code) is that non-orthogonal transforms are correctly applied to ellipsoids (see original issue).
there is a little bit of extraneous work done to plot spheres now - but not much as the slicing of the sphere was already handled by the same code that slices the ellipsoids.

Long term (after the release) we might want to think about deprecating the C++ spherical peak shape and just use ellipsoids but that would be a more major change for the next release.

**To test:**
(1) Follow instructions in the original issue - the non-orthog view of both the isotropic ellipsoids and sphere should be the same
(2) Add a background and check it still works in both orthog and non-orthog e.g. 
```
IntegratePeaksMD(InputWorkspace='ws', PeakRadius=0.6, BackgroundInnerRadius=0.6,
    BackgroundOuterRadius=0.8, PeaksWorkspace='peaks', OutputWorkspace='peaks_int_sphere', 
    Ellipsoid=False, UseOnePercentBackgroundCorrection=False)
IntegratePeaksMD(InputWorkspace='ws', PeakRadius=[0.6,0.6,0.6], BackgroundInnerRadius=0.6,
    BackgroundOuterRadius=0.8, PeaksWorkspace='peaks', OutputWorkspace='peaks_int_sphericalEllips', 
    Ellipsoid=True, UseOnePercentBackgroundCorrection=False)
```


Fixes #31275






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
